src/aktualizr_secondary/aktualizr_secondary.cc

```diff
@@ -16,7 +16,7 @@
 class SecondaryAdapter : public Uptane::SecondaryInterface {
  public:
   SecondaryAdapter(const Uptane::SecondaryConfig& sconfig_in, AktualizrSecondary& sec)
-      : SecondaryInterface(sconfig_in), secondary(sec) {}
```

Comment:
Does it make sense to pass `nullptr` as event queue? The main loop looks to be waiting for the event and won't send the report until it gets a response from everyone.

Reply:
SecondaryAdapter dont use event queue it is just a wrapper that uses real AktualizrSecondary implementation, passed as argument to the constructor

Reply:
Sorry, maybe I didn't quite get it. The loop at eventsinterpreter:38 will only proceed to `PutManifest` once it's collected as many `InstallComplete` events as there were installation requests. If some of the secondaries get `nullptr` for `events_channel_in` the interpreter will never get their `InstallComplete` and will aktualizr will basically hang waiting. Am I missing something?

Reply:
Yes, you are right, but what do you suggest? Throw exception if pointer passed to constructor is nullptr?

Reply:
Yes, either that or adding a method to the interface to tell if the secondary supports events (and event queue is not `nullptr`) and only increase ecu counter if it does.

Reply:
Also there is a work @patrickvacek is doing to deprecate eventsinterpreter, that probably should be coordinated.